lib/resolvers/union.js

```diff
@@ -9,14 +9,16 @@ import { getRecords } from "../orm/records";
  * @param {Object} args
  * @param {Object} context
  * @param {Object} info
+ * @param {Boolean} isList
+ * @param {Object} type
  * @see {@link https://graphql.org/learn/execution/#root-fields-resolvers}
  * @returns {Object[]} A list of records of many types from Mirage's database.
  */
-export default function resolveUnion(_obj, args, context, _info, type) {
+export default function resolveUnion(_obj, args, context, _info, isList, type) {
   const types = type.getTypes();
   const recordsForTypes = types.reduce(function (records, type) {
     return [...records, ...getRecords(type, args, context.mirageSchema)];
   }, []);
 
-  return recordsForTypes;
+  return isList ? recordsForTypes : recordsForTypes[0];
```

Comment:
Would `undefiend` being returned here break things if `recordsForTypes` returns an empty list?

Reply:
It shouldn't. If the field is required then yes but GraphQL should handle that and give you an appropriate error.